test/e2e/azure_privatecluster.go

```diff
@@ -329,4 +356,38 @@ func SetupExistingVNet(ctx context.Context, vnetCidr string, cpSubnetCidrs, node
 	Expect(err).To(BeNil())
 	err = vnetFuture.WaitForCompletionRef(ctx, vnetClient.Client)
 	Expect(err).To(BeNil())
+
+	return func() {
```

Comment:
Added a cleanup function to remove the resources we create that the cluster will not own, b/c the cluster will not try to delete resources it does not manage.